ocaml/typing/typedtree.mli

```diff
@@ -96,6 +109,7 @@ and 'a pattern_data =
     pat_type: Types.type_expr;
     pat_env: Env.t;
     pat_attributes: attributes;
+    pat_unique_barrier : Unique_barrier.t;
```

Comment:
It took us some work to discover what the semantics of this field is. Maybe capture this in a comment? Would it be right to say

```
(* This tracks whether the scrutinee of the pattern is used uniquely within the body of the pattern match. *)
```

?